dglgo/dglgo/apply_pipeline/graphpred/graphpred.jinja-py

```diff
@@ -0,0 +1,77 @@
+import torch
+import os
+import csv
+
+from tqdm import tqdm
+from dgl.data import AsGraphPredDataset
+from dgl.dataloading import GraphDataLoader
+{{ data_import_code }}
+
+{{ model_code }}
+
+def infer(device, loader, model):
+    model = model.to(device)
+    model.eval()
+    all_pred = []
+
+    with torch.no_grad():
+        for _, (g, labels) in enumerate(tqdm(loader, desc="Iteration")):
+            g = g.to(device)
+            node_feat = g.ndata['feat']
+            edge_feat = g.edata['feat']
+            pred = model(g, node_feat, edge_feat)
+            pred = (pred.sigmoid() >= 0.5).long()
+            all_pred.append(pred)
+
+    return torch.cat(all_pred, dim=0)
+
+def main():
+    {{ user_cfg_str }}
+
+    device = cfg['device']
+    if not torch.cuda.is_available():
+        device = 'cpu'
+    pipeline_cfg = cfg['general_pipeline']
+
+    # load data
+    data = AsGraphPredDataset({{data_initialize_code}})
+    data_loader = GraphDataLoader(data, batch_size=pipeline_cfg['batch_size'],
+                                  num_workers=pipeline_cfg['num_workers'], shuffle=False)
+
+    # validation
+    train_data_name = cfg['model']['data_info']['name']
+    infer_data_name = cfg['data']['name']
+    if train_data_name.startswith('ogbg-mol'):
+        assert infer_data_name.startswith('ogbg-mol'), 'Expect the inference data name to start \
+            with ogbg-mol, got {}'.format(infer_data_name)
+    else:
+        assert train_data_name == infer_data_name, 'Expect the training and inference data to \
+            have the same name, got {} and {}'.format(train_data_name, infer_data_name)
```

Comment:
Currently we only support ogbg-mol and CSV datasets. We can make it more flexible if we have more datasets in the future.

Reply:
Why only ogbg-mol? Is it because currently we only support binary classification?

Reply:
Generally, this might require supporting alternative encoders/model architectures (e.g., ogbg-code2), loss functions & evaluation metrics (multi-class classification, regression), outer loop (cross validation), etc.

Reply:
I see. Then can users choose datasets other than ogbg-mol and CSV via commandline? If not, I'm ok with the current design.

Reply:
No, the valid options only include `ogbg-molhiv`, `ogbg-molpcba`, and `csv`.